### PR TITLE
Install version dependent image streams for v1.0 and v1.1

### PR DESCRIPTION
**Pull request by sdodson**
This effectively means that v1.0 streams won't be updated from this point forward. If we want that then I think we need to maintain v1.0 content upstream.

FYI @bparees

### CODE REVIEW

**Comment by sdodson** [comment]
@detiber / @abutcher PTAL this works around image stream data incompatibility. This resets the v1-0 content to just before the last sync. Currently 3.0.2 and earlier environments fail to import the current image stream data.

**Comment by detiber** [comment]
@sdodson would it be better to use subdirs for v1.0 vs v1.1?  That way we could copy based on the subdirectory for the respective templates?

**Comment by sdodson** [comment]
@detiber Versioned directories sounds good to me. If we do that I'd say we should freeze v1.0 content as is and only updates to v1.1. I don't think we have the tools in place right now to test backwards comparability so that's the safest thing to do.

**Comment by detiber** [comment]
@sdodson :+1: If needed we can make changes as needed for v1.0 content as requested

**Comment by sdodson** [comment]
@detiber thanks for the suggestion that made the implementation much cleaner

File: roles/openshift_examples/examples-sync.sh [hunk 1]
```diff
@@ -6,9 +6,10 @@
 # This script should be run from openshift-ansible/roles/openshift_examples
 
 XPAAS_VERSION=ose-v1.1.0
-EXAMPLES_BASE=$(pwd)/files/examples
-find files/examples -name '*.json' -delete
-find files/examples -name '*.yaml' -delete
+ORIGIN_VERSION=v1.1
+EXAMPLES_BASE=$(pwd)/files/examples/${ORIGIN_VERSION}
+find files/examples${ORIGIN_VERSION} -name '*.json' -delete
+find files/examples${ORIGIN_VERSION} -name '*.yaml' -delete
```
Comment (detiber) [inline roles/openshift_examples/examples-sync.sh]: Should this be files/examples/${ORIGIN_VERSION} ?

Comment (detiber) [inline roles/openshift_examples/examples-sync.sh]: Also, couldn't you just use EXAMPLES_BASE here?

Comment (sdodson) [inline roles/openshift_examples/examples-sync.sh]: Yup, fixed now.

File: roles/openshift_examples/defaults/main.yml [hunk 2]
```diff
@@ -6,7 +6,9 @@ openshift_examples_load_db_templates: true
 openshift_examples_load_xpaas: "{{ openshift_deployment_type in ['enterprise','openshift-enterprise','atomic-enterprise','online']  }}"
 openshift_examples_load_quickstarts: true
 
-examples_base: /usr/share/openshift/examples
+content_version: "{{ 'v1.0' if not openshift.common.version_greater_than_3_1_or_1_1 else 'v1.1' }}"
```
Comment (detiber) [inline roles/openshift_examples/defaults/main.yml]: I think that switching this to: `"{{ 'v1.1' if openshift.common.version_greater_than_3_1_or_1_1 else 'v1.0' }}"` is a bit clearer.

Comment (sdodson) [inline roles/openshift_examples/defaults/main.yml]: I agree, updated.

**Comment by detiber** [comment]
:+1: from me

File: roles/openshift_examples/defaults/main.yml [hunk 3]
```diff
@@ -6,7 +6,9 @@ openshift_examples_load_db_templates: true
 openshift_examples_load_xpaas: "{{ openshift_deployment_type in ['enterprise','openshift-enterprise','atomic-enterprise','online']  }}"
 openshift_examples_load_quickstarts: true
 
-examples_base: /usr/share/openshift/examples
+content_version: "{{ 'v1.1' if openshift.common.version_greater_than_3_1_or_1_1 else 'v1.0' }}"
+
+examples_base: "/usr/share/openshift/examples"
```
Comment (abutcher) [inline roles/openshift_examples/defaults/main.yml]: Think this should be `examples_base: "/usr/share/openshift/{{ content_version }}"`

Comment (sdodson) [inline roles/openshift_examples/defaults/main.yml]: We're not altering the paths on the master, just the content source. Though I do see how that could be confusing. I'm open to landing all versions of content and then installing the appropriate version. If we go that route we'll need to update the docs.

Comment (abutcher) [inline roles/openshift_examples/defaults/main.yml]: Ah, path is altered now I believe - I got this error giving this a try:

```
TASK: [openshift_examples | Import RHEL streams] ******************************
failed: [master4.example.com] => {"changed": false, "cmd": ["oc", "create", "-n", "openshift", "-f", "/usr/share/openshift/examples/image-streams/image-streams-rhel7.json"], "delta": "0:00:00.167397", "end": "2015-11-18 14:43:24.730427", "failed": true, "failed_when_result": true, "rc": 1, "start": "2015-11-18 14:43:24.563030", "stdout_lines": [], "warnings": []}
stderr: the path "/usr/share/openshift/examples/image-streams/image-streams-rhel7.json" does not exist

FATAL: all hosts have already failed -- aborting
```

Comment (sdodson) [inline roles/openshift_examples/defaults/main.yml]: Were there any errors on the copy task? What do you have in /usr/share/openshift/examples/image-streams/ ?

Comment (abutcher) [inline roles/openshift_examples/defaults/main.yml]: Copy is good probably just wrong path on the dest:

```
TASK: [openshift_examples | Copy openshift examples] ************************** 
ok: [master4.example.com]

TASK: [openshift_examples | Import RHEL streams] ****************************** 
failed: [master4.example.com] => {"changed": false, "cmd": ["oc", "create", "-n", "openshift", "-f", "/usr/share/openshift/examples/image-streams/image-streams-rhel7.json"], "delta": "0:00:00.157610", "end": "2015-11-18 15:13:14.402957", "failed": true, "failed_when_result": true, "rc": 1, "start": "2015-11-18 15:13:14.245347", "stdout_lines": [], "warnings": []}
stderr: the path "/usr/share/openshift/examples/image-streams/image-streams-rhel7.json" does not exist
...
[root@master4 openshift-ansible]# ls -l /usr/share/openshift/
total 4
drwxr-xr-x. 8 root root 4096 Nov 18 14:43 v1.1
```

Comment (abutcher) [inline roles/openshift_examples/defaults/main.yml]: This does the right thing on my end:

```
 - name: Copy openshift examples
   copy:
-    src: "examples/{{ content_version }}"
-    dest: /usr/share/openshift
+    src: "examples/{{ content_version }}/"
+    dest: "{{ examples_base }}/"
```

Comment (sdodson) [inline roles/openshift_examples/defaults/main.yml]: Thanks, that fixed it for me too once i figured out my test host was dirty.

**Comment by abutcher** [comment]
Beep Boop Manual local vm test build/test passed for commit cef52c3 @brenton

**Comment by sdodson** [comment]
@brenton can we merge this? This will help people installing origin 1.0 and ose 3.0 from this repo

**Pull request merged**